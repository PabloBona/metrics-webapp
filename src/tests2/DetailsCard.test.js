import { render, screen } from '@testing-library/react';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: () => ({ id: 'bitcoin' }),
}));

jest.mock('../redux/slices/coinSlice', () => ({
  fetchCoin: jest.fn(),
}));

describe('CoinDetail Component', () => {
  test('renders loading when data is not yet available', () => {
    render(
      <div className="col">
        <h2 data-testid="h2test" className="align-items-center d-flex justify-content-center text-light vh-100">Loading...</h2>
      </div>,
    );
    const loadingElement = screen.getByTestId('h2test');
    expect(loadingElement).toHaveTextContent('Loading...');
    expect(loadingElement).toMatchSnapshot();
  });
});
