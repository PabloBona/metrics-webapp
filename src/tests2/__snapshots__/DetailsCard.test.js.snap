// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoinDetail Component renders loading when data is not yet available 1`] = `
<h2
  class="align-items-center d-flex justify-content-center text-light vh-100"
  data-testid="h2test"
>
  Loading...
</h2>
`;
